// vue-material-tabs/dist/vue-material-tabs.ssr.js
'use strict';function _typeof(obj) {
  "@babel/helpers - typeof";

  if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") {
    _typeof = function (obj) {
      return typeof obj;
    };
  } else {
    _typeof = function (obj) {
      return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj;
    };
  }

  return _typeof(obj);
}

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) {
  try {
    var info = gen[key](arg);
    var value = info.value;
  } catch (error) {
    reject(error);
    return;
  }

  if (info.done) {
    resolve(value);
  } else {
    Promise.resolve(value).then(_next, _throw);
  }
}

function _asyncToGenerator(fn) {
  return function () {
    var self = this,
        args = arguments;
    return new Promise(function (resolve, reject) {
      var gen = fn.apply(self, args);

      function _next(value) {
        asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value);
      }

      function _throw(err) {
        asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err);
      }

      _next(undefined);
    });
  };
}

function _slicedToArray(arr, i) {
  return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest();
}

function _toConsumableArray(arr) {
  return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread();
}

function _arrayWithoutHoles(arr) {
  if (Array.isArray(arr)) return _arrayLikeToArray(arr);
}

function _arrayWithHoles(arr) {
  if (Array.isArray(arr)) return arr;
}

function _iterableToArray(iter) {
  if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter);
}

function _iterableToArrayLimit(arr, i) {
  var _i = arr && (typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]);

  if (_i == null) return;
  var _arr = [];
  var _n = true;
  var _d = false;

  var _s, _e;

  try {
    for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) {
      _arr.push(_s.value);

      if (i && _arr.length === i) break;
    }
  } catch (err) {
    _d = true;
    _e = err;
  } finally {
    try {
      if (!_n && _i["return"] != null) _i["return"]();
    } finally {
      if (_d) throw _e;
    }
  }

  return _arr;
}

function _unsupportedIterableToArray(o, minLen) {
  if (!o) return;
  if (typeof o === "string") return _arrayLikeToArray(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor) n = o.constructor.name;
  if (n === "Map" || n === "Set") return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen);
}

function _arrayLikeToArray(arr, len) {
  if (len == null || len > arr.length) len = arr.length;

  for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i];

  return arr2;
}

function _nonIterableSpread() {
  throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}

function _nonIterableRest() {
  throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}function calc(e) {
  var offset = e.currentTarget.getBoundingClientRect();
  var localX = e.clientX - offset.left;
  var localY = e.clientY - offset.top;
  var radius = Math.sqrt(Math.pow(e.currentTarget.clientWidth, 2) + Math.pow(e.currentTarget.clientHeight, 2)) / 2;
  var left = "".concat(localX - radius, "px");
  var top = "".concat(localY - radius, "px");
  var size = "".concat(radius * 2, "px");
  return {
    left: left,
    top: top,
    size: size
  };
}

function mountRipple(e) {
  var _calc = calc(e),
      size = _calc.size,
      left = _calc.left,
      top = _calc.top;

  var rippleElement = document.createElement("span");
  rippleElement.classList.add("ripple");
  Object.assign(rippleElement.style, {
    height: size,
    width: size,
    top: top,
    left: left
  });
  e.currentTarget.appendChild(rippleElement);
  setTimeout(function () {
    return rippleElement.remove();
  }, 1000);
}

function bind$1(el, bidding) {
  bidding.value ? el.addEventListener("click", mountRipple) : el.removeEventListener("click", mountRipple);
}

function unbind$2(el) {
  el.removeEventListener("click", mountRipple);
}

var ripple = {
  bind: bind$1,
  update: bind$1,
  unbind: unbind$2
};function inserted(el, binding) {
  var _binding$value2, _binding$modifiers2;

  var timeoutId = 0;

  var onResize = function onResize(callback) {
    return function () {
      var _binding$value;

      clearTimeout(timeoutId);
      timeoutId = setTimeout(callback, Number((_binding$value = binding.value) === null || _binding$value === void 0 ? void 0 : _binding$value.delay) || 400);
    };
  };

  if (typeof binding.value === "function") {
    var _binding$modifiers;

    el._onResize = (_binding$modifiers = binding.modifiers) !== null && _binding$modifiers !== void 0 && _binding$modifiers.debounce ? onResize(binding.value) : binding.value;
  } else if (typeof ((_binding$value2 = binding.value) === null || _binding$value2 === void 0 ? void 0 : _binding$value2.cb) === "function" && (_binding$modifiers2 = binding.modifiers) !== null && _binding$modifiers2 !== void 0 && _binding$modifiers2.debounce) {
    el._onResize = onResize(binding.value.cb);
  }

  window.addEventListener("resize", el._onResize);
}

function unbind$1(el) {
  window.removeEventListener("resize", el._onResize);
  delete el._onResize;
}

var resize = {
  inserted: inserted,
  unbind: unbind$1
};function touchMove(event, el) {
  if (!el._onTouch.isSwiping) return;
  var touchendX = event.changedTouches[0].clientX;
  var minTouch = Math.abs(el.offsetWidth * 0.1);
  var diffX = el._onTouch.touchstartX - touchendX;

  var switchTo = function switchTo(to) {
    el._onTouch.isSwiping = false;

    el._onTouch.callback(to);
  };

  diffX > minTouch && switchTo("next");
  diffX < -minTouch && switchTo("prev");
}

function bind(el, binding) {
  if (typeof binding.value !== "function") return;
  el._onTouch = {
    callback: binding.value,
    isSwiping: false,
    touchstartX: 0,
    onTouchStart: function onTouchStart(event) {
      this._onTouch.isSwiping = true;
      this._onTouch.touchstartX = event.touches[0].clientX;
    },
    onTouchMove: function onTouchMove(event) {
      return touchMove(event, el);
    }
  };
  el.addEventListener("touchstart", el._onTouch.onTouchStart);
  el.addEventListener("touchmove", el._onTouch.onTouchMove);
}

function unbind(el) {
  if (!el._onTouch) return;
  el.removeEventListener("touchstart", el._onTouch.onTouchStart);
  el.removeEventListener("touchmove", el._onTouch.onTouchMove);
  delete el._onTouch;
}

var touch = {
  bind: bind,
  unbind: unbind
};//
//
//
//
//
//
//
//
var script$3 = {
  name: "Btn",
  inject: ["theme"]
};function normalizeComponent(template, style, script, scopeId, isFunctionalTemplate, moduleIdentifier /* server only */, shadowMode, createInjector, createInjectorSSR, createInjectorShadow) {
    if (typeof shadowMode !== 'boolean') {
        createInjectorSSR = createInjector;
        createInjector = shadowMode;
        shadowMode = false;
    }
    // Vue.extend constructor export interop.
    const options = typeof script === 'function' ? script.options : script;
    // render functions
    if (template && template.render) {
        options.render = template.render;
        options.staticRenderFns = template.staticRenderFns;
        options._compiled = true;
        // functional template
        if (isFunctionalTemplate) {
            options.functional = true;
        }
    }
    // scopedId
    if (scopeId) {
        options._scopeId = scopeId;
    }
    let hook;
    if (moduleIdentifier) {
        // server build
        hook = function (context) {
            // 2.3 injection
            context =
                context || // cached call
                    (this.$vnode && this.$vnode.ssrContext) || // stateful
                    (this.parent && this.parent.$vnode && this.parent.$vnode.ssrContext); // functional
            // 2.2 with runInNewContext: true
            if (!context && typeof __VUE_SSR_CONTEXT__ !== 'undefined') {
                context = __VUE_SSR_CONTEXT__;
            }
            // inject component styles
            if (style) {
                style.call(this, createInjectorSSR(context));
            }
            // register component module identifier for async chunk inference
            if (context && context._registeredComponents) {
                context._registeredComponents.add(moduleIdentifier);
            }
        };
        // used by ssr in case component is cached and beforeCreate
        // never gets called
        options._ssrRegister = hook;
    }
    else if (style) {
        hook = shadowMode
            ? function (context) {
                style.call(this, createInjectorShadow(context, this.$root.$options.shadowRoot));
            }
            : function (context) {
                style.call(this, createInjector(context));
            };
    }
    if (hook) {
        if (options.functional) {
            // register for functional component in vue file
            const originalRender = options.render;
            options.render = function renderWithStyleInjection(h, context) {
                hook.call(context);
                return originalRender(h, context);
            };
        }
        else {
            // inject component registration as beforeCreate hook
            const existing = options.beforeCreate;
            options.beforeCreate = existing ? [].concat(existing, hook) : [hook];
        }
    }
    return script;
}function createInjectorSSR(context) {
    if (!context && typeof __VUE_SSR_CONTEXT__ !== 'undefined') {
        context = __VUE_SSR_CONTEXT__;
    }
    if (!context)
        return () => { };
    if (!('styles' in context)) {
        context._styles = context._styles || {};
        Object.defineProperty(context, 'styles', {
            enumerable: true,
            get: () => context._renderStyles(context._styles)
        });
        context._renderStyles = context._renderStyles || renderStyles;
    }
    return (id, style) => addStyle(id, style, context);
}
function addStyle(id, css, context) {
    const group = css.media || 'default' ;
    const style = context._styles[group] || (context._styles[group] = { ids: [], css: '' });
    if (!style.ids.includes(id)) {
        style.media = css.media;
        style.ids.push(id);
        let code = css.source;
        style.css += code + '\n';
    }
}
function renderStyles(styles) {
    let css = '';
    for (const key in styles) {
        const style = styles[key];
        css +=
            '<style data-vue-ssr-id="' +
                Array.from(style.ids).join(' ') +
                '"' +
                (style.media ? ' media="' + style.media + '"' : '') +
                '>' +
                style.css +
                '</style>';
    }
    return css;
}/* script */
var __vue_script__$3 = script$3;
/* template */

var __vue_render__$3 = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('button', {
    staticClass: "vmt__btn",
    on: {
      "click": function click($event) {
        return _vm.$emit('click');
      }
    }
  }, [_vm._ssrNode("<svg viewBox=\"0 0 24 24\" role=\"img\" aria-hidden=\"true\"" + _vm._ssrStyle(null, {
    fill: _vm.theme.arrow
  }, null) + " data-v-c67981d6><path d=\"M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z\" data-v-c67981d6></path></svg>")]);
};

var __vue_staticRenderFns__$3 = [];
/* style */

var __vue_inject_styles__$3 = function __vue_inject_styles__(inject) {
  if (!inject) return;
  inject("data-v-c67981d6_0", {
    source: ".vmt__btn[data-v-c67981d6]{outline:0;border:none;background:0 0;cursor:pointer}.vmt__btn svg[data-v-c67981d6]{height:24px;width:24px;fill:rgba(214,213,213,.925)}.vmt__btn:disabled svg[data-v-c67981d6]{fill:#d6d6d652!important;cursor:default}",
    map: undefined,
    media: undefined
  });
};
/* scoped */


var __vue_scope_id__$3 = "data-v-c67981d6";
/* module identifier */

var __vue_module_identifier__$3 = "data-v-c67981d6";
/* functional template */

var __vue_is_functional_template__$3 = false;
/* style inject shadow dom */

var __vue_component__$3 = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$3,
  staticRenderFns: __vue_staticRenderFns__$3
}, __vue_inject_styles__$3, __vue_script__$3, __vue_scope_id__$3, __vue_is_functional_template__$3, __vue_module_identifier__$3, false, undefined, createInjectorSSR, undefined);var script$2 = {
  inject: ["theme"],
  components: {
    Btn: __vue_component__$3,
    VNode: {
      functional: true,
      render: function render(h, _ref) {
        var props = _ref.props;
        return props.node ? props.node : h("span", props.name);
      }
    }
  },
  directives: {
    ripple: ripple,
    resize: resize,
    touch: touch
  },
  props: {
    vertical: Boolean,
    navSlider: Boolean,
    ripple: Boolean,
    navAuto: Boolean,
    navItems: {
      type: Array,
      required: true
    },
    tabItemActive: {
      type: Object,
      default: function _default() {}
    }
  },
  data: function data() {
    return {
      pagination: {
        has: false,
        maxOffset: 0,
        minOffset: 0,
        translate: 0,
        offset: 0
      }
    };
  },
  computed: {
    classes: function classes() {
      return {
        vmt__nav: true,
        "vmt__nav--vertical": this.vertical,
        "vmt__nav--auto": this.navAuto && !this.vertical
      };
    },
    transform: function transform() {
      return "transform: translate".concat(this.vertical ? "Y" : "X", "(-").concat(this.pagination.translate, "px)");
    },
    paginateIndicator: function paginateIndicator() {
      var _this$pagination = this.pagination,
          translate = _this$pagination.translate,
          maxOffset = _this$pagination.maxOffset;
      return {
        next: translate < maxOffset,
        prev: translate > 0
      };
    },
    orientation: function orientation() {
      return this.vertical ? "portrait" : "landscape";
    },
    sliderElement: function sliderElement() {
      var _this$$refs;

      return (_this$$refs = this.$refs) === null || _this$$refs === void 0 ? void 0 : _this$$refs.slider;
    }
  },
  watch: {
    navItems: "resizable",
    // Force recalc the pagination offsets
    // when the orientation/navItems is change;
    vertical: function vertical() {
      Object.assign(this.$data, this.$options.data());
      this.resizable();
    },
    tabItemActive: function tabItemActive(_tabItemActive) {
      this.setSliderPosition();

      if (this.pagination.has && _tabItemActive !== null && _tabItemActive !== void 0 && _tabItemActive.model) {
        var _navItemActiveElement, _navElement$getClient;

        var navItemActiveElement = this.$refs[_tabItemActive.model][0];
        var navElement = this.$refs.nav;
        var navActiveItemRectsCollection = navItemActiveElement === null || navItemActiveElement === void 0 ? void 0 : (_navItemActiveElement = navItemActiveElement.getClientRects()) === null || _navItemActiveElement === void 0 ? void 0 : _navItemActiveElement[0];
        var navRectsCollection = navElement === null || navElement === void 0 ? void 0 : (_navElement$getClient = navElement.getClientRects()) === null || _navElement$getClient === void 0 ? void 0 : _navElement$getClient[0];
        if (!navActiveItemRectsCollection || !navRectsCollection) return;
        var _this$pagination2 = this.pagination,
            translate = _this$pagination2.translate,
            maxOffset = _this$pagination2.maxOffset;
        var toTranslate = translate; // Portrait

        if (this.orientation === "portrait") {
          if (navActiveItemRectsCollection.bottom > navRectsCollection.bottom) {
            toTranslate = toTranslate + navActiveItemRectsCollection.height;
          }

          if (navActiveItemRectsCollection.top < navRectsCollection.top) {
            toTranslate = navActiveItemRectsCollection.height > toTranslate ? 0 : toTranslate - navActiveItemRectsCollection.height;
          }
        }

        if (this.orientation === "landscape") {
          if (navActiveItemRectsCollection.right > navRectsCollection.right) {
            toTranslate = toTranslate + navActiveItemRectsCollection.width;
          }

          if (navActiveItemRectsCollection.left < navRectsCollection.left) {
            toTranslate = navActiveItemRectsCollection.width > toTranslate ? 0 : toTranslate - navActiveItemRectsCollection.width;
          }

          if (toTranslate > maxOffset) {
            toTranslate = toTranslate + (maxOffset - toTranslate);
          }
        }

        this.pagination.translate = Math.abs(toTranslate);
      }
    }
  },
  mounted: function mounted() {
    var _this$theme = this.theme,
        nav = _this$theme.nav,
        navItem = _this$theme.navItem,
        slider = _this$theme.slider;
    this.$el.style.background = nav;
    this.$refs.navItems.style.color = navItem;
    if (this.sliderElement) this.sliderElement.style.background = slider;
  },
  methods: {
    select: function select(navItem) {
      this.$emit("select", navItem);
    },
    setSliderPosition: function setSliderPosition() {
      var _this = this;

      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee() {
        var _this$tabItemActive, _this$$refs$navItems, _this$$refs$navItems$, _this$$refs$model$, _this$$refs$model$$ge;

        var model, navItemsRectsCollection, navActiveItemRectsCollection, _this$sliderElement;

        return regeneratorRuntime.wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                _context.next = 2;
                return _this.$nextTick();

              case 2:
                model = (_this$tabItemActive = _this.tabItemActive) === null || _this$tabItemActive === void 0 ? void 0 : _this$tabItemActive.model;

                if (model) {
                  _context.next = 5;
                  break;
                }

                return _context.abrupt("return");

              case 5:
                navItemsRectsCollection = (_this$$refs$navItems = _this.$refs.navItems) === null || _this$$refs$navItems === void 0 ? void 0 : (_this$$refs$navItems$ = _this$$refs$navItems.getClientRects()) === null || _this$$refs$navItems$ === void 0 ? void 0 : _this$$refs$navItems$[0];
                navActiveItemRectsCollection = (_this$$refs$model$ = _this.$refs[model][0]) === null || _this$$refs$model$ === void 0 ? void 0 : (_this$$refs$model$$ge = _this$$refs$model$.getClientRects()) === null || _this$$refs$model$$ge === void 0 ? void 0 : _this$$refs$model$$ge[0];

                if (_this.sliderElement) {
                  Object.assign((_this$sliderElement = _this.sliderElement) === null || _this$sliderElement === void 0 ? void 0 : _this$sliderElement.style, {
                    portrait: {
                      height: navActiveItemRectsCollection.height + "px",
                      top: navActiveItemRectsCollection.top - navItemsRectsCollection.top + "px",
                      width: "",
                      left: ""
                    },
                    landscape: {
                      width: navActiveItemRectsCollection.width + "px",
                      left: navActiveItemRectsCollection.left - navItemsRectsCollection.left + "px",
                      height: "",
                      top: ""
                    }
                  }[_this.orientation]);
                }

              case 8:
              case "end":
                return _context.stop();
            }
          }
        }, _callee);
      }))();
    },
    setPaginationSettings: function setPaginationSettings() {
      var _this$$refs2;

      var paginationRefactory = function paginationRefactory(has, maxOffset, minOffset) {
        return {
          has: has,
          maxOffset: maxOffset,
          minOffset: minOffset,
          offset: minOffset
        };
      };

      var navElement = this.$refs.nav;
      var navElementWidth = navElement === null || navElement === void 0 ? void 0 : navElement.offsetWidth;
      var navElementHeight = navElement === null || navElement === void 0 ? void 0 : navElement.offsetHeight;
      var navItemsElement = (_this$$refs2 = this.$refs) === null || _this$$refs2 === void 0 ? void 0 : _this$$refs2.navItems;
      var navItemsElementWidth = navItemsElement === null || navItemsElement === void 0 ? void 0 : navItemsElement.offsetWidth;

      var navItemsElementHeight = _toConsumableArray(navItemsElement === null || navItemsElement === void 0 ? void 0 : navItemsElement.children).slice(0, -1).reduce(function (previousValue, currentValue) {
        return previousValue + currentValue.offsetHeight;
      }, 0);

      Object.assign(this.pagination, {
        portrait: paginationRefactory(navItemsElementHeight > navElementHeight, navItemsElementHeight - navElementHeight, navElementHeight),
        landscape: paginationRefactory(navItemsElementWidth > navElementWidth, navItemsElementWidth - navElementWidth, navElementWidth)
      }[this.orientation]);

      if (this.pagination.maxOffset === 0) {
        this.pagination.translate = 0;
      }
    },
    paginationTo: function paginationTo(to) {
      var _this$pagination3 = this.pagination,
          maxOffset = _this$pagination3.maxOffset,
          offset = _this$pagination3.offset,
          translate = _this$pagination3.translate,
          minOffset = _this$pagination3.minOffset;

      if (to === "prev" && this.paginateIndicator.prev) {
        if (offset <= minOffset) {
          this.pagination.offset = minOffset;
        }

        if (translate - offset < offset) {
          this.pagination.translate = 0;
          return;
        }

        this.pagination.translate = translate - offset;
      }

      if (to === "next" && this.paginateIndicator.next) {
        if (translate + offset > maxOffset) {
          this.pagination.offset = maxOffset - translate;
        }

        this.pagination.translate = translate + this.pagination.offset;
      }
    },
    resizable: function resizable() {
      var _this2 = this;

      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee2() {
        return regeneratorRuntime.wrap(function _callee2$(_context2) {
          while (1) {
            switch (_context2.prev = _context2.next) {
              case 0:
                _context2.next = 2;
                return _this2.$nextTick();

              case 2:
                _this2.setPaginationSettings();

                _this2.setSliderPosition();

              case 4:
              case "end":
                return _context2.stop();
            }
          }
        }, _callee2);
      }))();
    },
    getActiveColor: function getActiveColor(_ref2) {
      var model = _ref2.model;

      if (model === this.tabItemActive.model) {
        return {
          color: this.theme.navActiveItem
        };
      }
    }
  }
};/* script */
var __vue_script__$2 = script$2;
/* template */

var __vue_render__$2 = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('div', {
    directives: [{
      name: "resize",
      rawName: "v-resize.debounce",
      value: _vm.resizable,
      expression: "resizable",
      modifiers: {
        "debounce": true
      }
    }],
    class: _vm.classes
  }, [_vm._ssrNode("<div class=\"vmt__nav__prev\" data-v-7d287132>", "</div>", [_vm.pagination.has ? _c('Btn', {
    attrs: {
      "disabled": !_vm.paginateIndicator.prev
    },
    on: {
      "click": function click($event) {
        return _vm.paginationTo('prev');
      }
    }
  }) : _vm._e()], 1), _vm._ssrNode(" "), _c('nav', {
    directives: [{
      name: "touch",
      rawName: "v-touch",
      value: _vm.paginationTo,
      expression: "paginationTo"
    }],
    ref: "nav",
    staticClass: "tab__nav"
  }, [_vm._ssrNode("<ul class=\"vmt__nav__items\"" + _vm._ssrStyle(null, _vm.transform, null) + " data-v-7d287132>", "</ul>", [_vm._l(_vm.navItems, function (navItem) {
    return _c('li', {
      directives: [{
        name: "ripple",
        rawName: "v-ripple",
        value: _vm.ripple && !navItem.disabled,
        expression: "ripple && !navItem.disabled"
      }],
      key: navItem.model,
      ref: navItem.model,
      refInFor: true,
      staticClass: "vmt__nav__item",
      class: {
        active: navItem.model === _vm.tabItemActive.model,
        disabled: navItem.disabled
      },
      style: _vm.getActiveColor(navItem),
      on: {
        "click": function click($event) {
          $event.preventDefault();
          return _vm.select(navItem);
        }
      }
    }, [_c('VNode', {
      attrs: {
        "node": navItem.nameSlot,
        "name": navItem.name
      }
    })], 1);
  }), _vm._ssrNode(" " + (_vm.navSlider ? "<hr class=\"vmt__tab__slider\" data-v-7d287132>" : "<!---->"))], 2)]), _vm._ssrNode(" "), _vm._ssrNode("<div class=\"vmt__nav__next\" data-v-7d287132>", "</div>", [_vm.pagination.has ? _c('Btn', {
    attrs: {
      "disabled": !_vm.paginateIndicator.next
    },
    on: {
      "click": function click($event) {
        return _vm.paginationTo('next');
      }
    }
  }) : _vm._e()], 1)], 2);
};

var __vue_staticRenderFns__$2 = [];
/* style */

var __vue_inject_styles__$2 = function __vue_inject_styles__(inject) {
  if (!inject) return;
  inject("data-v-7d287132_0", {
    source: ".vmt__nav[data-v-7d287132]{display:flex;justify-content:space-between;align-items:center;vertical-align:middle;max-width:100%;flex:0 1 auto;position:relative;contain:content}.vmt__nav .vmt__nav__prev[data-v-7d287132],.vmt__nav__next[data-v-7d287132]{flex:1 40px;min-width:40px}.vmt__nav__next[data-v-7d287132] .vmt__btn svg{transform:rotate(180deg)}.vmt__nav nav[data-v-7d287132]{position:relative;display:flex;overflow:hidden;flex:1 100%}.vmt__nav__items[data-v-7d287132]{position:relative;display:flex;margin:0;padding:0;flex:1 auto;transition:.3s cubic-bezier(.25,.8,.5,1);height:100%}.vmt__nav__items .vmt__nav__item[data-v-7d287132]{list-style:none;text-align:center;cursor:pointer;padding:.9rem 1rem;letter-spacing:.0892857143em;display:flex;justify-content:center;align-items:center;text-align:center;text-transform:uppercase;font-size:.875rem;font-weight:500;white-space:normal;transition:background .1s ease;position:relative;overflow:hidden;min-width:90px;max-width:360px;user-select:none}.vmt__nav__items .vmt__nav__item[data-v-7d287132]:hover:not(.disabled,.active){background:hsla(0,0%,100%,.09)}.vmt__nav__items .active[data-v-7d287132]:hover{background:hsla(0,0%,100%,.18)}.vmt__nav__items .disabled[data-v-7d287132]{background:#6969694f}.vmt__tab__slider[data-v-7d287132]{height:2px;width:2px;border:none;margin:0;padding:0;bottom:0;position:absolute;transition:left .3s cubic-bezier(.25,.8,.5,1),top .3s cubic-bezier(.25,.8,.5,1)}.vmt__nav--vertical[data-v-7d287132]{flex-direction:column}.vmt__nav--vertical .vmt__nav__items[data-v-7d287132]{flex-direction:column;flex:1 auto;position:relative}.vmt__nav--vertical .vmt__nav__item *[data-v-7d287132]{padding:0;margin:0}.vmt__nav--vertical[data-v-7d287132] .vmt__nav__prev svg{transform:rotate(90deg)}.vmt__nav--vertical[data-v-7d287132] .vmt__nav__next svg{transform:rotate(270deg)}.vmt__nav--vertical .vmt__nav__item[data-v-7d287132]{justify-content:left;padding-top:1.6rem;padding-bottom:1.6rem}.vmt__nav--auto .vmt__nav__item[data-v-7d287132]{flex:1 auto}",
    map: undefined,
    media: undefined
  }), inject("data-v-7d287132_1", {
    source: ".ripple{z-index:2;background-color:hsla(0,0%,100%,.23);border-radius:50%;position:absolute;transform:scale(0);animation:ripple .6s linear}@keyframes ripple{to{transform:scale(2.5);opacity:0}}",
    map: undefined,
    media: undefined
  });
};
/* scoped */


var __vue_scope_id__$2 = "data-v-7d287132";
/* module identifier */

var __vue_module_identifier__$2 = "data-v-7d287132";
/* functional template */

var __vue_is_functional_template__$2 = false;
/* style inject shadow dom */

var __vue_component__$2 = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$2,
  staticRenderFns: __vue_staticRenderFns__$2
}, __vue_inject_styles__$2, __vue_script__$2, __vue_scope_id__$2, __vue_is_functional_template__$2, __vue_module_identifier__$2, false, undefined, createInjectorSSR, undefined);var themeFactory = function themeFactory(nav, navItem, navActiveItem, slider, arrow) {
  return {
    nav: nav,
    navItem: navItem,
    navActiveItem: navActiveItem,
    slider: slider,
    arrow: arrow
  };
};

var $themes = {
  default: themeFactory("#fff", "#000000", "#1867c0", "#1867c0", "#616161"),
  purple: themeFactory("#4A148C", "#BDBDBD", "#fff", "#CE93D8", "#f3f3f3"),
  red: themeFactory("#F44336", "#f3f3f3", "#fff", "#EF9A9A", "#f3f3f3"),
  pink: themeFactory("#E91E63", "#F8BBD0", "#fff", "#F8BBD0", "#f3f3f3"),
  cyan: themeFactory("#00BCD4", "#B2EBF2", "#fff", "#80DEEA", "#f3f3f3"),
  green: themeFactory("#4CAF50", "#C8E6C9", "#fff", "#A5D6A7", "#f3f3f3")
};var script$1 = {
  name: "Tabs",
  components: {
    TabNav: __vue_component__$2
  },
  directives: {
    touch: touch
  },
  provide: function provide() {
    return {
      theme: this.getTheme
    };
  },
  props: {
    value: [Number, String],
    theme: {
      type: [Object, String],
      default: "default"
    },
    vertical: Boolean,
    ripple: {
      type: Boolean,
      default: true
    },
    slideDuration: {
      type: [String, Number],
      default: 200
    },
    slideVertical: Boolean,
    slide: {
      type: Boolean,
      default: true
    },
    navAuto: Boolean,
    navSlider: {
      type: Boolean,
      default: true
    },
    noTouch: Boolean
  },
  data: function data() {
    return {
      slideSide: "",
      navItems: [],
      tabItemActive: {
        model: "",
        name: "",
        disabled: false
      },
      tabItemIndexes: {
        current: 0,
        last: 0,
        previous: 0
      },
      tabItems: []
    };
  },
  computed: {
    classes: function classes() {
      return {
        tabs: true,
        "tabs--vertical": this.vertical
      };
    },
    transition: function transition() {
      return {
        slide: this.slide,
        duration: this.slide ? this.slideDuration : 0,
        vertical: this.slideVertical,
        side: this.slideSide
      };
    },
    getTheme: function getTheme() {
      if (typeof this.theme === "string") {
        return $themes[this.theme] || $themes.default;
      } else if (_typeof(this.theme) === "object" && this.theme) {
        return this.theme;
      }

      return $themes.default;
    }
  },
  watch: {
    tabItemActive: function tabItemActive(newTabItem, oldTabItem) {
      this.tabItemIndexes.previous = this.finTabIndex(oldTabItem);
      this.tabItemIndexes.current = this.finTabIndex(newTabItem);
      if (!(newTabItem !== null && newTabItem !== void 0 && newTabItem.model)) return;
      this.tabItems[this.tabItemIndexes.current].activeModel = newTabItem.model; // eslint-disable-next-line no-extra-boolean-cast

      if (Boolean(~this.tabItemIndexes.previous)) {
        this.setTabItemTransitionSide();
        this.tabItems[this.tabItemIndexes.previous].activeModel = newTabItem.model;
      }
    },
    value: {
      immediate: true,
      handler: function handler() {
        var _this = this;

        var value = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 0;
        this.$nextTick(function () {
          var _this$navItems;

          var navItem = (_this$navItems = _this.navItems) === null || _this$navItems === void 0 ? void 0 : _this$navItems[value];
          navItem && _this.activeTabItem(navItem);
        });
      }
    }
  },
  methods: {
    register: function register(tabItemInstance) {
      if (this.isTabItemComponent(tabItemInstance)) {
        this.tabItems.push(tabItemInstance);
        this.setNavItem(tabItemInstance);
      }
    },
    unRegister: function unRegister(tabItemInstance) {
      if (this.isTabItemComponent(tabItemInstance)) {
        this.disableTabItem(tabItemInstance.ownNavItemIndex);
        this.tabItems.splice(this.finTabIndex(tabItemInstance), 1);
        this.navItems.splice(tabItemInstance.ownNavItemIndex, 1);
        this.tabItemIndexes.last = this.navItems.length - 1;
      }
    },
    isTabItemComponent: function isTabItemComponent(_ref) {
      var $options = _ref.$options;
      return ($options === null || $options === void 0 ? void 0 : $options.name) === "TabItem";
    },
    setNavItem: function setNavItem(_ref2) {
      var _$slots$name;

      var model = _ref2.model,
          name = _ref2.name,
          disabled = _ref2.disabled,
          $slots = _ref2.$slots;
      this.navItems.push({
        model: model,
        name: name,
        disabled: disabled,
        nameSlot: (_$slots$name = $slots.name) === null || _$slots$name === void 0 ? void 0 : _$slots$name[0]
      });
      this.tabItemIndexes.last = this.navItems.length - 1;
    },
    activeTabItem: function activeTabItem(tabItem) {
      if (!(tabItem !== null && tabItem !== void 0 && tabItem.disabled)) {
        this.tabItemActive = tabItem;
        this.$emit("input", this.finTabIndex(tabItem));
      }
    },
    disableTabItem: function disableTabItem(tabItemIndex) {
      var _this$tabItemIndexes = this.tabItemIndexes,
          current = _this$tabItemIndexes.current,
          last = _this$tabItemIndexes.last;

      if (tabItemIndex === current) {
        var nextTabItem = current === last ? current - 1 : current + 1;
        this.activeTabItem(this.navItems[nextTabItem]);
      }
    },
    setTabItemTransitionSide: function setTabItemTransitionSide() {
      var _this$tabItemIndexes2 = this.tabItemIndexes,
          current = _this$tabItemIndexes2.current,
          previous = _this$tabItemIndexes2.previous;
      this.slideSide = current > previous ? "right" : "left";
    },
    switchTabItem: function switchTabItem(to) {
      var _this2 = this;

      if (!this.noTouch) {
        var _this$tabItemIndexes3 = this.tabItemIndexes,
            current = _this$tabItemIndexes3.current,
            last = _this$tabItemIndexes3.last;

        var switchTab = function switchTab(count) {
          var _this2$navItems;

          var tabItem = (_this2$navItems = _this2.navItems) === null || _this2$navItems === void 0 ? void 0 : _this2$navItems[current + count];
          tabItem && _this2.activeTabItem(tabItem);
        };

        to === "next" && current < last && switchTab(1);
        to === "prev" && current > 0 && switchTab(-1);
      }
    },
    finTabIndex: function finTabIndex(tab) {
      return this.tabItems.findIndex(function (el) {
        return el.model === (tab === null || tab === void 0 ? void 0 : tab.model);
      });
    }
  }
};/* script */
var __vue_script__$1 = script$1;
/* template */

var __vue_render__$1 = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('div', {
    class: _vm.classes
  }, [_vm._t("nav", [_c('TabNav', _vm._b({
    on: {
      "select": _vm.activeTabItem
    }
  }, 'TabNav', {
    ripple: _vm.ripple,
    navAuto: _vm.navAuto,
    navItems: _vm.navItems,
    vertical: _vm.vertical,
    navSlider: _vm.navSlider,
    tabItemActive: _vm.tabItemActive
  }, false))], {
    "items": _vm.navItems,
    "active": _vm.tabItemActive
  }), _vm._ssrNode(" "), _c('div', {
    directives: [{
      name: "touch",
      rawName: "v-touch",
      value: _vm.switchTabItem,
      expression: "switchTabItem"
    }],
    staticClass: "tabs__content"
  }, [_vm._t("default")], 2)], 2);
};

var __vue_staticRenderFns__$1 = [];
/* style */

var __vue_inject_styles__$1 = function __vue_inject_styles__(inject) {
  if (!inject) return;
  inject("data-v-1d334e66_0", {
    source: ".tabs[data-v-1d334e66]{background:#fff;display:flex;flex-direction:column;border-radius:.23rem;height:100%;width:100%}.tabs__content[data-v-1d334e66]{display:flex;position:relative;justify-content:center;align-items:center;overflow:hidden}.tabs--vertical[data-v-1d334e66]{flex-direction:row}",
    map: undefined,
    media: undefined
  });
};
/* scoped */


var __vue_scope_id__$1 = "data-v-1d334e66";
/* module identifier */

var __vue_module_identifier__$1 = "data-v-1d334e66";
/* functional template */

var __vue_is_functional_template__$1 = false;
/* style inject shadow dom */

var __vue_component__$1 = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$1,
  staticRenderFns: __vue_staticRenderFns__$1
}, __vue_inject_styles__$1, __vue_script__$1, __vue_scope_id__$1, __vue_is_functional_template__$1, __vue_module_identifier__$1, false, undefined, createInjectorSSR, undefined);//
//
//
//
//
//
//
//
var script = {
  name: "TabItem",
  props: {
    name: {
      type: String,
      default: "Tab Item"
    },
    disabled: Boolean
  },
  data: function data() {
    return {
      activeModel: "",
      model: Math.random().toString("16").substring("2")
    };
  },
  mounted: function mounted() {
    this.$parent.register(this);
  },
  beforeDestroy: function beforeDestroy() {
    this.$parent.unRegister(this);
  },
  computed: {
    transition: function transition() {
      return this.$parent.transition || 200;
    },
    isActive: function isActive() {
      return this.activeModel === this.model && !this.disabled;
    },
    slideDirection: function slideDirection() {
      if (!this.transition.slide) return "";

      if (this.transition.vertical) {
        return {
          left: "slide-top",
          right: "slide-bottom"
        }[this.transition.side];
      }

      return {
        left: "slide-left",
        right: "slide-right"
      }[this.transition.side];
    },
    ownNavItemIndex: function ownNavItemIndex() {
      var _this$$parent,
          _this = this;

      return (_this$$parent = this.$parent) === null || _this$$parent === void 0 ? void 0 : _this$$parent.navItems.findIndex(function (navItem) {
        return navItem.model === _this.model;
      });
    }
  },
  watch: {
    disabled: function disabled(payload) {
      var _this$$parent2;

      payload && ((_this$$parent2 = this.$parent) === null || _this$$parent2 === void 0 ? void 0 : _this$$parent2.disableTabItem(this.ownNavItemIndex));
      this.$parent.navItems[this.ownNavItemIndex].disabled = payload;
    }
  },
  methods: {
    beforeEnterHandle: function beforeEnterHandle(element) {
      element.style.position = "relative";
    },
    leaveHandle: function leaveHandle(element) {
      element.style.position = "absolute";
    }
  }
};/* script */
var __vue_script__ = script;
/* template */

var __vue_render__ = function __vue_render__() {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('transition', {
    attrs: {
      "name": _vm.slideDirection
    },
    on: {
      "beforeEnter": _vm.beforeEnterHandle,
      "leave": _vm.leaveHandle
    }
  }, [_c('div', {
    directives: [{
      name: "show",
      rawName: "v-show",
      value: _vm.isActive,
      expression: "isActive"
    }],
    staticClass: "vmt__tab-item",
    style: {
      'transition-duration': _vm.transition.duration + "ms"
    }
  }, [_vm._t("default")], 2)]);
};

var __vue_staticRenderFns__ = [];
/* style */

var __vue_inject_styles__ = function __vue_inject_styles__(inject) {
  if (!inject) return;
  inject("data-v-5683d9d1_0", {
    source: ".vmt__tab-item[data-v-5683d9d1]{top:0;left:0;z-index:1;width:100%;height:inherit;background:#fff;transition:transform cubic-bezier(.25,.8,.5,1)}.slide-left-enter[data-v-5683d9d1],.slide-right-leave-to[data-v-5683d9d1]{transform:translateX(-100%)}.slide-left-leave-to[data-v-5683d9d1],.slide-right-enter[data-v-5683d9d1]{transform:translateX(100%)}.slide-bottom-leave-to[data-v-5683d9d1],.slide-top-enter[data-v-5683d9d1]{transform:translateY(-100%)}.slide-bottom-enter[data-v-5683d9d1],.slide-top-leave-to[data-v-5683d9d1]{transform:translateY(100%)}",
    map: undefined,
    media: undefined
  });
};
/* scoped */


var __vue_scope_id__ = "data-v-5683d9d1";
/* module identifier */

var __vue_module_identifier__ = "data-v-5683d9d1";
/* functional template */

var __vue_is_functional_template__ = false;
/* style inject shadow dom */

var __vue_component__ = /*#__PURE__*/normalizeComponent({
  render: __vue_render__,
  staticRenderFns: __vue_staticRenderFns__
}, __vue_inject_styles__, __vue_script__, __vue_scope_id__, __vue_is_functional_template__, __vue_module_identifier__, false, undefined, createInjectorSSR, undefined);var components$1=/*#__PURE__*/Object.freeze({__proto__:null,Tabs: __vue_component__$1,TabItem: __vue_component__});var install = function installVueMaterialTabs(Vue) {
  Object.entries(components$1).forEach(function (_ref) {
    var _ref2 = _slicedToArray(_ref, 2),
        componentName = _ref2[0],
        component = _ref2[1];

    Vue.component(componentName, component);
  });
};var components=/*#__PURE__*/Object.freeze({__proto__:null,'default': install,Tabs: __vue_component__$1,TabItem: __vue_component__});// only expose one global var, with component exports exposed as properties of
// that global var (eg. plugin.component)

Object.entries(components).forEach(function (_ref) {
  var _ref2 = _slicedToArray(_ref, 2),
      componentName = _ref2[0],
      component = _ref2[1];

  if (componentName !== 'default') {
    install[componentName] = component;
  }
});module.exports=install;
